internal/group.go

```diff
@@ -318,9 +344,17 @@ func (g *group) run(specs Specs) {
 				g.suite.currentSpecReport.CapturedGinkgoWriterOutput += string(g.suite.writer.Bytes())
 				g.suite.currentSpecReport.CapturedStdOutErr += g.suite.outputInterceptor.StopInterceptingAndReturnOutput()
 
-				if g.suite.currentSpecReport.State.Is(types.SpecStatePassed | types.SpecStateSkipped | types.SpecStateAborted | types.SpecStateInterrupted) {
-					break
+				switch t := multipleExecutionDecorator; {
+				case t == reflect.TypeOf(FlakeAttempts(0)):
+					if g.suite.currentSpecReport.State.Is(types.SpecStatePassed | types.SpecStateSkipped | types.SpecStateAborted | types.SpecStateInterrupted) {
+						break maxAttemptForLoop
+					}
+				case t == reflect.TypeOf(RepeatAttempts(0)):
+					if g.suite.currentSpecReport.State.Is(types.SpecStateFailed | types.SpecStateSkipped | types.SpecStateAborted | types.SpecStateInterrupted) {
```

Comment:
What about `types.SpecStatePanicked`?

Reply:
yes.  in fact... I think you want `if !g.suite.currentSpecReport.State.Is(types.SpecStatePassed) {` - since the idea is it has to pass cleanly each time.

Reply:
In that case ` if g.suite.currentSpecReport.State.Is(types.SpecStateFailureStates) {` might be cleaner

Reply:
sure that works too :)